vonk/deployment/performance.rst

```diff
@@ -0,0 +1,161 @@
+
+Pipeline
+~~~~~~~~
+
+Vonk is made up of a pipeline of components. You can leave out any component that you don't need. So if you don't need conditional processing (create, update, delete), just exclude them from the pipeline. Excluded components are not loaded and thus never executed. See :ref:`settings_pipeline`.
+
+Platform
+^^^^^^^^
+
+Vonk can run on Windows, Linux and MacOS. Directly or in a Docker container. On real hardware, virtual machine, app service or Kubernetes cluster. And then you can choose the dimensions of the platform, scaling up (bigger 'hard'ware) or scaling out (more (virtual) machines) as you see fit. Each of these choices will influence performance.
+
+Besides the way Vonk is deployed, the way the database is deployed is an important factor. Vonk needs a very low latency connection between the Vonk process(es) and the database server. If you have configured :ref:`configure_log_insights`, the calls to the database are recorded as separate dependencies so you can check whether this may be a bottleneck.
+
+Vonk is optimized for multithreaded processing. This means that it will benefit from extra processing cores, either in the same machine (multi core processor) or by adding additional machines (and thus processors). 
+
+Vonk is fully stateless, so no data about connections is saved between requests. First of all this helps in scaling out, since you don't need thread affinity. On top of that this limits the memory requirements of Vonk.
+
+Usage patterns
+^^^^^^^^^^^^^^
+
+How will Vonk be used in your environment? 
+
+#. Mostly for querying or rather for creating and updating resources?
+   Altering resources requires more processing than reading them. Also see the comment on indexing and search parameters above.
+#. How is the distribution of values in the resources that you query on?
+   E.g. if you use only a few types of resources, query them just by tag and the resources have only about 5 different tags, calculating the number of results will take a lot of time. Using more finegrained distributed values to query on solves this.
+#. With many individual resources or with (large) batches or transactions?
+   Transactions take a lot longer to process and require more memory, proportionally to the number of resources in them. If many transactions are run in parallel, requests may queue up. 
+#. Many users with a low request rate each, or a few heavy users? 
+   Since Vonk is stateless, this has little influence. The total request rate is what counts. 
+
+Testing performance yourself
+----------------------------
+
+Because of all the variables mentioned above the best way to find out whether Vonk's performance is sufficient for your use is: Test it yourself.
```

Comment:
```suggestion
Because of all the variables mentioned above the best way to find out whether Vonk's performance is sufficient for your use is: test it yourself.
```